src/Core/Extension.php

```diff
@@ -44,6 +44,13 @@
      */
     private $ownerStack = [];
 
+    public $class;
+
+    public function __construct()
+    {
+        $this->class = get_class($this);
```

Comment:
you'll need to add parent::__construct() to all the places it was removed, as well.

Also, we've removed the $class field, that shouldn't be re-introduced.